File: src/contests/cg/puzzle/java/1d-bush-fire.java
// 🎮 CodinGame Puzzle - 1d-bush-fire
// https://www.codingame.com/training/easy/1d-bush-fire

import java.io.*;

class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());
        for (int t = 0; t < n; t++) {
            String strip = br.readLine().replaceAll("\\s+$", "");
            int drops = 0, j = 0;
            while (j < strip.length()) {
                if (strip.charAt(j) == 'f') {
                    // Drop at j covers j, j+1, j+2 — skip past all 3.
                    drops++;
                    j += 3;
                } else {
                    j++;
                }
            }
            System.out.println(drops);
        }
    }
}

File: src/contests/cg/puzzle/fsharp/1d-bush-fire.fs
// 🎮 CodinGame Puzzle - 1d-bush-fire
// https://www.codingame.com/training/easy/1d-bush-fire

let n = int (System.Console.ReadLine().Trim())
for _ in 1 .. n do
    let strip = System.Console.ReadLine().TrimEnd()
    let mutable drops = 0
    let mutable j = 0
    while j < strip.Length do
        if strip.[j] = 'f' then
            // Drop at j covers j, j+1, j+2 — skip past all 3.
            drops <- drops + 1
            j <- j + 3
        else
            j <- j + 1
    printfn "%d" drops
